### PR TITLE
Document how to develop and run the code with maximum caching

### PR DESCRIPTION
**Pull request by PhilMiller**

### CODE REVIEW

**Review by peterscherpelz — APPROVED** [review]
@PhilMiller I went ahead and linked the development page in both the index and from the main README, just making the changes directly. Someone else should review my changes before we merge though.

File: mewarpx/docs/development.rst
Comment (roelof-groenewald) [inline mewarpx/docs/development.rst]: `minerva` is not a dependency for `mewarpx` and since this is all public, I'd say we should remove this line:
```suggestion
```

**Review by roelof-groenewald — CHANGES_REQUESTED** [review]
This is great, thanks Phil! One other request besides the one suggested: Could you please revert the change in `Source/Make.WarpX` line 236 where I earlier changed the `pywarpx` install to be in development mode since that change is no longer needed.

**Review by KZhu-ME — APPROVED** [review]
Overall looks good, I agree with Roelof about leaving out the minerva install step. Otherwise, everything looks good to merge once that's taken care of.

File: mewarpx/docs/development.rst
Comment (PhilMiller) [reply mewarpx/docs/development.rst]: Will push with the comment corrected, too.

**Review by roelof-groenewald — APPROVED** [review]
Thanks!

**Pull request merged**
